src/main/java/persistence/sql/ddl/metadata/TableMetadata.java

```diff
@@ -0,0 +1,33 @@
+package persistence.sql.ddl.metadata;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import persistence.utils.StringUtils;
+
+public class TableMetadata {
+    private final String name;
+
+    private TableMetadata(String name) {
+        this.name = name;
+    }
+
+    public static TableMetadata from(Class<?> clazz) {
+        if (!clazz.isAnnotationPresent(Entity.class)) {
+            throw new IllegalArgumentException("@Entity가 아닙니다");
+        }
+        String name = extractName(clazz);
+        return new TableMetadata(name);
+    }
+
+    private static String extractName(Class<?> clazz) {
+        if (clazz.isAnnotationPresent(Table.class) && !clazz.getDeclaredAnnotation(Table.class).name().isEmpty()) {
+            return clazz.getDeclaredAnnotation(Table.class).name();
+        }
+
+        return StringUtils.convertToSnakeCase(clazz.getSimpleName());
+    }
```

Comment:
```suggestion
    private static String extractName(Class<?> clazz) {
        final Table annotation = clazz.getDeclaredAnnotation(Table.class);
        if (annotation == null || annotation.name().isEmpty()) {
            return StringUtils.convertToSnakeCase(clazz.getSimpleName());
        }

        return annotation.name();
    }
```

조건문이 조금 복잡해보이는데, 이런 느낌의 코드는 어떠신가요 ㅎㅎ